ethereum/sidecar/server.go

```diff
@@ -953,11 +981,106 @@ func (s *Server) fetchAssetsUnlockConfirmedEvents(
 	return abiEvents, nil
 }
 
-//nolint:unparam
-func (s *Server) attestAssetsUnlockedEvents(ctx context.Context) error {
-	// TODO: Implement
-	<-ctx.Done()
-	return nil
+func (s *Server) unqueueAttestation() *bridgetypes.AssetsUnlockedEvent {
+	s.attestationMutex.Lock()
+	defer s.attestationMutex.Unlock()
+	if len(s.attestationQueue) == 0 {
+		return nil
+	}
+
+	attestation := s.attestationQueue[0]
+	s.attestationQueue = s.attestationQueue[1:]
+
+	return &attestation
+}
+
+func (s *Server) attestAssetsUnlockedEvents(ctx context.Context) {
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			// get attestations if any
+			for attestation := s.unqueueAttestation(); attestation != nil; attestation = s.unqueueAttestation() {
+
+				// try to execute this transaction until we have done so successfully.
+				retry := true
+				for retry && s.shouldAttest(attestation) {
```

Comment:
Let's also extend `shouldAttest` with a call to `validateAssetsUnlocked` to make sure we are attesting a valid entry.